test/expect.js

```diff
@@ -1129,11 +1129,16 @@ describe('expect', function () {
         lenFn  = function() { return heroes.length };
 
     expect(fn).to.change(obj, 'value');
+    expect(fn).to.change(obj, 'value').by(5);
+
     expect(sameFn).to.not.change(obj, 'value');
+    expect(sameFn).to.not.change(obj, 'value').by(1);
+
     expect(sameFn).to.not.change(obj, 'str');
     expect(bangFn).to.change(obj, 'str');
-    expect(batFn).to.change(lenFn);
-    expect(sameFn).to.not.change(lenFn);
+
+    expect(batFn).to.change(lenFn).by(1);
+    expect(sameFn).to.not.change(lenFn).by(1);
```

Comment:
Really seems like a situation we should get covered.

About the error cases: you mean trying to use `by` alongside an assertion other than `change`, `increases` and `decreases` or testing the error it throws when the delta doesn't match?